lib/tlaw/params/type.rb

```diff
@@ -91,23 +97,32 @@ def initialize(enum)
           else
             fail ArgumentError, "Unparseable enum: #{enum.inspect}"
           end
+        )
       end
 
-      def values
-        type.keys
+      def possible_values
+        type.keys.map(&:inspect).join(', ')
       end
 
-      def validate(value)
-        type.key?(value) or
-          nonconvertible!(
-            value,
-            "is not one of #{type.keys.map(&:inspect).join(', ')}"
-          )
+      def validation_error(value)
+        "is not one of #{possible_values}" unless type.key?(value)
       end
 
       def _convert(value)
         type[value]
       end
     end
+
+    # @private
+    class RangeType < Type
+      def possible_values
+        type.inspect
+      end
+
+      def validation_error(value)
+        return 'is not an Integer' if type.begin.is_a?(Integer) && !value.is_a?(Integer)
```

Comment:
This seem surprisingly narrow. Care to explain?..

Reply:
If the range is `1..42`, then only integer values would be acceptable. If the range is `1.0..42.0`, then float or integers would be ok. I don't need the floating range personally, but I thought supporting it this way made sense.

Reply:
Hm ok, let's leave it this way for now. My intuition says it means we need to somehow separate "expected type" and "expected range" from range, but it is non-trivial (because naive solution would prohibit integer for float ranges, for example), so may be I am overthinking. So let's clarify this thing with `validate_error` in neighbour PR, and then merge.